### ISSUE TITLE
Multiple ParserElements with same result name

### ISSUE THREAD

**Issue by Nicholas42** [pyparsing/pyparsing#205]
If I have multiple `ParserElement`s with the same result name I would expect one of two possible outcomes:

1. All matched expressions will be available as as list for that key.
2. Only the last one will be available.

Consider the following code:

```python3
import pyparsing as pp

word = pp.Word(pp.printables)("word")

res = pp.OneOrMore(word).parseString("test blub")

print(repr(res))
# prints:    (['test', 'blub'], {'word': ['test', 'blub']})
print(res["word"])
# prints:    blub
print(res.asDict())
# prints:    {'word': 'blub'}
```

As you see in the output of the first `print` the `ParseResult` saves both matched words as value for `"word"`, that would match outcome 1. But if I try to access this list by `res2["word"]`, I only get the last word back. This seems inconsistent to me. 

In my opinion, the `repr` and the return value of `res.asDict()` should look the same. So either `repr(res)` returns `repr(res.asDict())` (as `res.dump()` does) or `res["word"]` returns `['test', 'blub']`. Personally I would prefer the second variant, since it gives more freedom to the user and allows collecting results from very different parts of the string. 

PS: One can access the list by the (a bit illegal) `res._ParseResults__tokdict["word"]`. If you don't think that it would be a good idea to change the behavior of `ParseResults`, one maybe could make this variable publically accessible. 

PPS: I love the library :-)

**Comment by ptmcg** [comment]
The repr output is incorrect, at first glance that looks to me like a bug.

If you want to get all the matches, then add an asterisk to the end of your results name like `word = pp.Word(pp.printables)("word*")` (equivalent to calling `setResultsName` with `listAllMatches=True`).

**Comment by Nicholas42** [comment]
Ah, thx. That was exactly what I needed.

**Comment by ptmcg** [comment]
Released in 3.0.0a2
